content/tracing/setup/first_class_dimensions.md

```diff
@@ -0,0 +1,151 @@
+
+## Environment
+### Definition
+
+An environment is a first-class dimension that is used to scope an entire Datadog APM application. Some display settings are shared across environments, but all the measurable data (traces/metrics/statistics) can not be re-aggregated across multiple environments. Use cases include:
+
+* Stage environments such as production, staging, and pre-production...
+
+Environments are [tags][1], therefore they must follow the following rules:
+
+* They must start with a lower case letter.
+* Other characters must be alphanumeric lower case Unicode characters, underscores, minuses, colons, periods or slashes.
+* They must not be more than 100 characters long.
+
+Environments in traces and configuration files are normalized:
+
+* Unsupported characters are replaced by underscores.
+* Upper case characters are converted to lower case.
+
+### Default environment
+
+The default environment for un-tagged data is `env:none`. See below to see how to specify custom environments:
+
+### Setup
+
+There are several ways to specify an environment when reporting data:
+
+1. Host tag:  
+  Use a host tag with the format `env:XXXX` to tag all traces from that Agent accordingly.
+
+2. Agent configuration:  
+  Override the default tag used by the trace Agent in [the Agent configuration file][2]. This tags all traces coming through the Agent, overriding the host tag value.
+
+    ```
+    apm_config:
+      env: pre-prod
+    ```
+
+3. Per trace:  
+  When submitting a single trace, specify an environment by tagging one of its spans with the metadata key `env`. This overrides the Agent configuration and the host tags value (if any).  
+
+  * **Go**:
+    ```
+    tracer.SetTag("env", "prod")
+    ```
+  For OpenTracing use the `tracer.WithGlobalTag` start option to set the environment globally.
+
+  * **Java**:  
+      Via sysprop: 
+      ```
+      -Ddd.trace.span.tags=”env:prod”
+      ```
+      Via env var:
+      ```
+      DD_TRACE_SPAN_TAGS=”env:prod”
+      ```
+
+  * **Ruby**:
+  ```
+  Datadog.tracer.set_tags(‘env’ => ‘prod’)
+  ```
+
+  * **Python**:
+    ```
+    from ddtrace import tracer
+    tracer.set_tags('env', 'prod')
+    ```
+
+### Viewing Data by Environment
+
+Environments appear near the top of APM pages. Use the dropdown to scope the data displayed on the current page.
+
+{{< img src="tracing/setup/first_class_dimensions/envs_tracing_screen.png" alt="Envs tracing" responsive="true" popup="true" style="width:80%;">}}
+
+
+## Primary Tags
+### Definition
+
+A primary tag is a first-class dimension that is used to scope an entire Datadog APM application. Primary tags are used in conjunction with environments to get an even finer view of your application’s behavior. Use cases for primary tags include:
+
+* Availability zone
+* Datacenter
+
+A primary tag must follow these rules:
```

Comment:
If primary tags have different rules from an ordinary tag, I would say it explicitly. "Primary tags must follow a different set of rules from those of conventional tags." If they don't have special rules then it should sort of be the same as the tag rules above.

Reply:
+1

Reply:
They do have a different set of rules.